Form/ChoosePaymentMethodType.php

```diff
@@ -161,11 +161,11 @@ public function configureOptions(OptionsResolver $resolver)
             'choice_options'  => 'array',
         );
 
-        if (Legacy::supportsFormTypeConfigureOptions()) {
+        if (Legacy::supportsOptionsResolverSetAllowedTypesAsArray()) {
             $resolver->setAllowedTypes($allowedTypes);
         } else {
             foreach ($allowedTypes as $key => $value) {
-                $resolver->addAllowedTypes($key, $value);
+                $resolver->setAllowedTypes($key, $value);
```

Comment:
`setAllowedTypes` is always available, so I thought not confusing people by changing that would be best.